File: zad1&2/Tipper/Tipper/app/src/main/java/com/example/tipper/activities/QuizController.java
package com.example.tipper.activities;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;
import java.util.Random;

public class QuizController {

    private List<QuizQuestion> questionsAndAnswers = new ArrayList<>();
    private List<String> hardcodedQuestionsThatShouldBeTakenFromResources = new ArrayList<>();
    private int currentQuestionIndex = 0;
    private int goodAnswers = 0;
    private boolean isFinished = false;

    public List<QuizQuestion> getQuestionsAndAnswers() {
        return questionsAndAnswers;
    }

    public boolean isFinished() {
        return isFinished;
    }

    public int getGoodAnswers() {
        return goodAnswers;
    }

    public QuizController() {
        GenerateQuestions();
        ShuffleQuestions();
    }

    public String GetCurrentQuestion(){
        return questionsAndAnswers.get(currentQuestionIndex).getQuestion();
    }

    public String GetCurrentAnswer1(){
        return questionsAndAnswers.get(currentQuestionIndex).getAnswer1();
    }

    public String GetCurrentAnswer2(){
        return questionsAndAnswers.get(currentQuestionIndex).getAnswer2();
    }

    public void Answer(int answerIndex){
        if(questionsAndAnswers.get(currentQuestionIndex).getGoodAnswerIndex() == answerIndex)
        {
            goodAnswers++;
        }

        currentQuestionIndex++;
        if(currentQuestionIndex >= questionsAndAnswers.size()){
            FinishQuiz();
        }
    }

    private void FinishQuiz() {
        isFinished = true;
    }

    private void GenerateQuestions() {
        hardcodedQuestionsThatShouldBeTakenFromResources.add("Jedzenie fastfoodów i przetworzonego jedzenia jest zdrowe.;Prawda;Fałsz;1");
        hardcodedQuestionsThatShouldBeTakenFromResources.add("Zjedzenie jednego dużego posiłku jest zdrowsze niż kilku mniejszych w ciągu dnia.;Prawda;Fałsz;0");
        hardcodedQuestionsThatShouldBeTakenFromResources.add("Ile powinno się pić litrów wody dziennie?;Około 2l;Wcale, lepsze są napoje z dużą zawartością cukru;0");
        hardcodedQuestionsThatShouldBeTakenFromResources.add("Który z wymienionych produktów zawiera najwięcej witaminy C na 100g produktu?;Chrzan;Czarna porzeczka;1");
        hardcodedQuestionsThatShouldBeTakenFromResources.add("Stosując dietę ubogą w tłuszcze narażamy się na niedobory witamin. Jakich?;A,D,E,K;C#,C++,C,R;0");
        hardcodedQuestionsThatShouldBeTakenFromResources.add("Czy jedzenie kurzych jaj jest zdrowe?;Tak;Tak, ale tylko w odpowiednich ilościach;1");

        for (int i=0;i<hardcodedQuestionsThatShouldBeTakenFromResources.size(); i++)
        {
            String question = hardcodedQuestionsThatShouldBeTakenFromResources.get(i).split(";")[0];
            String answer1 = hardcodedQuestionsThatShouldBeTakenFromResources.get(i).split(";")[1];
            String answer2 = hardcodedQuestionsThatShouldBeTakenFromResources.get(i).split(";")[2];
            int goodAnswerIndex = Integer.parseInt(hardcodedQuestionsThatShouldBeTakenFromResources.get(i).split(";")[3]);
            QuizQuestion quizQuestion = new QuizQuestion(question, answer1, answer2, goodAnswerIndex);
            questionsAndAnswers.add(quizQuestion);
        }
    }

    private void ShuffleQuestions(){
        Collections.shuffle(questionsAndAnswers);
    }


}

File: zad1&2/Tipper/Tipper/app/src/main/java/com/example/tipper/activities/QuizController.kt
package com.example.tipper.activities

import android.content.Context
import android.content.res.Resources
import com.example.tipper.R
import java.util.*

class QuizController(val applicationContext: Context) {
    private val questionsAndAnswers: MutableList<QuizQuestion?> = ArrayList()
    private val questionsFromRes: MutableList<String> = ArrayList()
    private var currentQuestionIndex = 0
    private var goodAnswers = 0
    private var isFinished = false


    fun isFinished(): Boolean {
        return isFinished
    }

    fun getGoodAnswers(): Int {
        return goodAnswers
    }

    fun getQuestionsAndAnswers(): MutableList<QuizQuestion?>? {
        return questionsAndAnswers
    }

    fun GetCurrentQuestion(): String? {
        return questionsAndAnswers[currentQuestionIndex]!!.getQuestion()
    }

    fun GetCurrentAnswer1(): String? {
        return questionsAndAnswers[currentQuestionIndex]!!.getAnswer1()
    }

    fun GetCurrentAnswer2(): String? {
        return questionsAndAnswers[currentQuestionIndex]!!.getAnswer2()
    }

    fun Answer(answerIndex: Int) {
        if (questionsAndAnswers[currentQuestionIndex]!!.getGoodAnswerIndex() == answerIndex) {
            goodAnswers++
        }
        currentQuestionIndex++
        if (currentQuestionIndex >= questionsAndAnswers.size) {
            FinishQuiz()
        }
    }

    private fun FinishQuiz() {
        isFinished = true
    }

    private fun GenerateQuestions() {
        questionsFromRes.add("${applicationContext.getString(R.string.question1)};${applicationContext.getString(R.string.truth)};${applicationContext.getString(R.string.not_truth)};1")
        questionsFromRes.add("${applicationContext.getString(R.string.question2)};${applicationContext.getString(R.string.truth)};${applicationContext.getString(R.string.not_truth)};0")
        questionsFromRes.add("${applicationContext.getString(R.string.question3)};${applicationContext.getString(R.string.q3_truth)};${applicationContext.getString(R.string.q3_not_truth)};0")
        questionsFromRes.add("${applicationContext.getString(R.string.question4)};${applicationContext.getString(R.string.q4_not_truth)};${applicationContext.getString(R.string.q4_truth)};1")
        questionsFromRes.add("${applicationContext.getString(R.string.question5)};${applicationContext.getString(R.string.q5_truth)};${applicationContext.getString(R.string.q5_not_truth)};0")
        questionsFromRes.add("${applicationContext.getString(R.string.question6)};${applicationContext.getString(R.string.q6_not_truth)};${applicationContext.getString(R.string.q6_truth)};1")
        for (i in questionsFromRes.indices) {
            val question = questionsFromRes[i].split(";").toTypedArray()[0]
            val answer1 = questionsFromRes[i].split(";").toTypedArray()[1]
            val answer2 = questionsFromRes[i].split(";").toTypedArray()[2]
            val goodAnswerIndex = questionsFromRes[i].split(";").toTypedArray()[3].toInt()
            val quizQuestion = QuizQuestion(question, answer1, answer2, goodAnswerIndex)
            questionsAndAnswers.add(quizQuestion)
        }
    }

    private fun ShuffleQuestions() {
        Collections.shuffle(questionsAndAnswers)
    }

    init {
        GenerateQuestions()
        ShuffleQuestions()
    }
}